src/components/04-tables/tables.jsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * JSX Tables Component Attributes
+ * @function Tables
+ * @param {Object} config - The configuration object for table attributes
+ * @property {string} config.name  - The name of the table.
+ * @property {string} config.class - The classes inherited from the USWDS CSS files. The default class is `usa-table` and additions can be appended.
+ * @property {Array} config.children - When this template is used to make a react component @babel/core will change "config" to "props" so that children components can be accessed.
+ * @property {string} config.id    - The ID of the table
+ * @example
+ *   const config = {
+ *     name: "Table",
+ *     class: "usa-table--borderless",
+ *     id: "813"
+ *   };
+ *   Tables(config)
+ * @returns {HTMLTableElement} The HTML Table
+ */
+exports.Tables = function (config) {
+  return (
+    <table id={`${config.tableID}`} class={`usa-table ${config.class}`}>
+      <caption>{config.caption}</caption>
+      {config.children}
+    </table>
+  );
+};
+
+/**
+ * @function TableHeader - function that returns HTML code for a table header
+ * @param {Object} config - configuration object for tables
+ * @property {Array} config.header - The array of data for the column header.
+ * @example
+ *  const config = {
+ *   header: [
+ *       "Document title",
+ *       "Description",
+ *       "Year"
+ *     ]
+ * }
+ * TableHeader(config);
+ * @returns {HTMLTableHeaderCellElement} - The HTML for the table header
+ */
+exports.TableHeader = function (config) {
+  let tableHeader = [];
+
+  for (let i = 0; i < config.header.length; i++) {
+    tableHeader.push(`<th scope='col'>${config.header[i]}</th>`);
+  }
+
+  return (
+    <thead>
+      <tr>{tableHeader.join("\n")}</tr>
+    </thead>
+  );
+};
+/**
+ * @function TableRow - function that returns HTML code for a table row
+ * @param {Object} config - configuration object for tables
+ * @property {Array} config.row - The 2D array of strings that define each row and cell within that row.
+ * @example
+ * const config = {
+ *  row: [
+ *       [
+ *         "Declaration of Independence",
+ *         "Statement adopted by the...",
+ *         "1776",
+ *       ],
+ *       [
+ *         "Bill of Rights",
+ *         "The first ten amendments of the U.S...",
+ *         "1791"
+ *       ],
+ *       [
+ *         "Row-n Column-1",
+ *         "Row-n Column-2",
+ *         "Row-n Column-3"
+ *       ],
+ *     ]
+ * };
+ * TableRow(config);
+ * @returns {HTMLTableRowElement} - The HTML for the table row
+ */
+exports.TableRow = function (config) {
```

Comment:
Consider either making this a single TableRow (i.e. no `for` loop, and let the developer to that part) or name it plural TableRows. Naming conventions generally should follow the input and/or output of the function.

As a side note, and definitely not a blocker, I would generally lean towards making this a Singular function and then as a user of the React version I would implement it as:

```
<Table name="Table" id="demo">
  <TableHeader header=["foo", "bar", "baz"]/>
  {tableRows.map(row => <TableRow row={row.data}/>)}
</Table>
```

In this case my tableRows would look like:
```
tableRows = [
  ["foo", "bar", "baz"],
  ["oof", "rab", "zab"]
]
```

This would let you simplify your code, while making it a bit more clear to the end dev what is going on.

Reply:
I changed the **TableRow** function so that it only returns 1 `<tr>` element so its not confusing with its name. I also made it so a class can be passed into `<th>` just like you mentioned with the **TableHeader**.